crates/nu-cli/src/cli.rs

```diff
@@ -724,14 +727,39 @@ async fn process_line(
             // ...and we're in the CLI
             // ...then change to this directory
             if cli_mode && pipeline.commands.list.len() == 1 {
-                if let ClassifiedCommand::External(ExternalCommand {
+                if let ClassifiedCommand::Internal(InternalCommand {
                     ref name, ref args, ..
                 }) = pipeline.commands.list[0]
                 {
-                    if dunce::canonicalize(&name).is_ok()
+                    let internal_name = name;
+                    let name = args
+                        .positional
+                        .as_ref()
+                        .and_then(|potionals| {
+                            potionals.get(0).map(|e| {
+                                if let Expression::Literal(Literal::String(ref s)) = e.expr {
+                                    &s
+                                } else {
+                                    ""
+                                }
+                            })
+                        })
+                        .unwrap_or("");
+
+                    if internal_name == "run_external"
+                        && args
+                            .positional
+                            .as_ref()
+                            .map(|ref v| v.len() == 1)
+                            .unwrap_or(true)
+                        && args
+                            .named
+                            .as_ref()
+                            .map(NamedArguments::is_empty)
+                            .unwrap_or(true)
+                        && dunce::canonicalize(&name).is_ok()
```

Comment:
All of the above is rather unfortunate, but I think this is a bit of a hack right now.